### PR TITLE
feat: detect children usage automatically

### PR DESCRIPTION
**Pull request by marionebl**
## Proposed changes

Currently Alva requires users to type `children` explicitly. This creates adoption costs we want to avoid, thus we'll detect as much `props.children` usage as possible automatically by analyzing the component implementation.

* [x] Unit testing setup
* [x] Proof of concept for `React.SFC`
* [x] `export const Component =`
* [x] `export function Component`
* [x] `export class Component`
* [x] Aliasing
* [x] Destructuring
* [x] Integration

### Accept children automatically if used

**Status**: ✅ 

  <details>
  <summary>Steps</summary>

  * 1. Open a new file
  * 2. `git clone -b chore-remove-explicit-children-props --depth=1 --single-branch https://github.com/meetalva/designkit.git`
  * 3. Hit "Connect New Library"
  * 4. Select package.json in cloned designkit
  * 5. See e.g. `Button` accepting `children` without explicit types
  * 6. See e.g. `Space` not accepting `children`

  </details>

### CODE REVIEW

**Comment by marionebl** [comment]
Deployed at: https://alva-issues-689.surge.sh

File: src/analyzer/fixtures/children.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	return <div>{children}</div>;
+};
+
+export const AliasingSFC: React.SFC = props => {
+	const p = props;
+	const c = p.children;
+	return <div>{c}</div>;
+};
+
+export class DestrucutingClassComponent extends React.Component {
```
Comment (tilmx) [inline src/analyzer/fixtures/children.tsx]: DestrucutingClassComponent -> DestructuringClassComponent

**Comment by Palmaswell** [comment]
```
export class Copy extends React.Component<CopyProps> {
	public render () {
		return (
		<StyledCopy
			className={this.props.className}
			tagName={this.props.tagName}
			size={this.props.size}
			textAlign={this.props.textAlign}
			color={this.props.color}
			uppercase={this.props.uppercase}
			>
			{this.props.children}
		</StyledCopy>
		);
	}
}
```

```
export class Copy extends React.Component<CopyProps> {
	render() {
		return (
			<div>
				{this.props.children}
			</div>
		)
	}
}
```

**Comment by marionebl** [comment]
@Palmaswell I adressed the issue and added appropriate tests to the utility I factored out in the process.

**Pull request merged**